### PR TITLE
Identity - add browser customization options

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    public BrowserCustomizationOptions setHtmlMessageSuccess(String htmlMessageSuccess) {
+        this.htmlMessageSuccess = htmlMessageSuccess;
+        return this;
+    }
+
+    /**
+     * Configure the property to set HtmlMessageError which the browser will show to the user when the user
+     * finishes authenticating successfully, but an error occurred. You can use a string format e.g.
+     * "An error has occurred: {0} details: {1}."
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java:28]: Where does the detail for the format string come from? Is there another doc we can link to?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @return the string message.
+     */
+    public String getHtmlMessageSuccess() {
+        return this.htmlMessageSuccess;
+    }
+
+    /**
+     * Get the configured message which the browser will show to the user when the user
+     * finishes authenticating successfully, but an error occurred.
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java:50]: does "authenticating successfully but an error occurred" mean that the uesr's credentials were valid but no token was available? It's a confusing statement to me right now. I also wonder if this is a possible state if there needs to be another state for "authentication failed" ?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java [hunk 3]
```diff
@@ -209,4 +219,6 @@ public InteractiveBrowserCredential build() {
         return new InteractiveBrowserCredential(clientId, tenantId, port, redirectUrl, automaticAuthentication,
             loginHint, identityClientOptions);
     }
+
+
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java:223]: remove

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java [hunk 4]
```diff
@@ -465,7 +467,7 @@ OnBehalfOfParameters buildOBOFlowParameters(TokenRequestContext request) {
         return builder.build();
     }
 
-    InteractiveRequestParameters.InteractiveRequestParametersBuilder buildInteractiveRequestParameters(TokenRequestContext request, String loginHint, URI redirectUri) {
+    InteractiveRequestParameters.InteractiveRequestParametersBuilder buildInteractiveRequestParameters(TokenRequestContext request, String loginHint, URI redirectUri, BrowserCustomizationOptions browserCustomizationOptions) {
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java:470]: small nit: `IdentityClientBase` has access to the options, so do we need the parameter here?

Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java:470]: small nit: `IdentityClientBase` has access to the options, so do we need the parameter here?

**Review by billwert — APPROVED** [review]
Approving but I would like to see you remove the blank lines. The rest are questions or comments that we can discuss and I'm not super particular about.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java (hunk 4 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java]: refactored.

Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java]: refactored.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java (hunk 2 above)
Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/BrowserCustomizationOptions.java]: removed successfully from the text.

**Pull request merged**